packages/select-menu/src/types.ts

```diff
@@ -0,0 +1,43 @@
+export type SelectMenuAppearanceType = 'primary' | 'minimal' | 'card'
+
+export type SelectMenuChildrenFn<T extends OptionBase> = <T>(props: {
+  toggle: () => void
+  getRef: (ref: any) => void
+  isShown: boolean
+  selectedItems: T | T[]
+}) => React.ReactNode
+
+export interface SelectMenuProps<T extends OptionBase> {
+  options: T[]
+  value: string | string[]
+  children: React.ReactNode | SelectMenuChildrenFn<T>
+  appearance: SelectMenuAppearanceType
+  hasFilter?: boolean
+  hasTitle?: boolean
+  title?: string
+  width?: number
+  height?: number
+  hasCloseButton: boolean
+  onSelect: (value: string) => void
+  onDeselect: (value: string) => void
+  multiOptionSelectedItemsLabel?: (itemsSelectedLength: number) => string
+  isMultiSelect: boolean
+  compareBy: string
```

Comment:
Please review which props are required and which are not. I'm pretty sure that this can be optional. Some of other props also looks like might be optional